index.bs

```diff
@@ -617,6 +710,52 @@ SessionResult = (StatusResult)
 
 </pre>
 
+<div algorithm>
+
+To <dfn lt="updating the event map">update the event map</dfn>, given
```

Comment:
I'm thinking about implementations. For CDP, when a subscribe command is processed, we'll need to actually enable the appropriate CDP domain(s) before updating the event map and completing the command. I'm sure Gecko and WebKit would need to do something similar.

My concern is that the spec as written doesn't guarantee that the underlying platform (CDP, Gecko, WebKit, etc) agrees with the event map before returning from the subscribe command. It might be beneficial to add a step to this algorithm; Something like:

- Take implementation-defined steps to ensure the events listed in `event names` are enabled for the context with id `context id`.

Reply:
I don't think this really makes sense spec-wise. I agree that implementations are going to have to do that, but the obvious way to specify this is that you write steps which are triggered by hooks in other specs. Those other steps always run regardless of WebDriver BiDi spec. Then in our even dispatch algorithm we check if the event is enabled given an event name and a browsing context name. 

In a real implementation of course you don't want to process every event and then check at the last moment whether it's enabled for the browsing context, you want to process the minimum possible number of events only for browsing contexts where they will be used, and that probably means updating internal state at this point. But I think trying to work out how to write that down in a spec is complicated and probably not the most useful use of our time. I guess I can add a note.